src/main/scala/com/microsoft/sqlserver/jdbc/spark/SQLServerBulkJdbcOptions.scala

```diff
@@ -1,24 +1,31 @@
 package com.microsoft.sqlserver.jdbc.spark
 
 import java.sql.Connection
+
 import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
-import org.apache.spark.sql.execution.datasources.jdbc.JdbcOptionsInWrite
+import org.apache.spark.sql.execution.datasources.jdbc.{JDBCOptions, JdbcOptionsInWrite}
 
 class SQLServerBulkJdbcOptions(val params: CaseInsensitiveMap[String]) extends JdbcOptionsInWrite(params) {
 
     def this(params: Map[String, String]) = this(CaseInsensitiveMap(params))
-    
+
     // Save original parameters for when a JdbcBulkOptions instance is passed
     // from the Spark driver to an executor, which loses the reference to the
     // params input in memory
     override val parameters = params
 
+    val dbtable = params.getOrElse("dbtable", null)
+    val databaseName = params.getOrElse("databaseName", null)
+
     val user = params.getOrElse("user", null)
     val password = params.getOrElse("password", null)
-    val dbtable = params.getOrElse("dbtable", null)
 
+  //  AAD Authentication
+    val accessToken = params.getOrElse("accessToken", null)
+    val encrypt = params.getOrElse("encrypt", null)
```

Comment:
Where are these parameters used? Clearly i am missing something there, was hoping to see these parameters passed in connection creation.

Reply:
These are used under the hood in Spark.  If you follow the trail in the upstream classes, JDBCOptions, you'll see they basically get passed over to the specified driver.  In this case, its SQLServerDriver:  `ACCESS_TOKEN("accessToken", ""),`

I believe the options pass-thru up the chain via _spark\sql\core\src\main\scala\org\apache\spark\sql\execution\datasources\jdbc\connection\BasicConnectionProvider.scala_

```
import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions

private[jdbc] class BasicConnectionProvider(driver: Driver, options: JDBCOptions)
    extends ConnectionProvider {
  def getConnection(): Connection = {
    val properties = getAdditionalProperties()
    options.asConnectionProperties.entrySet().asScala.foreach { e =>
      properties.put(e.getKey(), e.getValue())
    }
    driver.connect(options.url, properties)
  }
}
```
Our options class, `SQLServerJDBCBulkOptions`, extends JDBCOptionsInWrite which is an implementation class for the companion class JDBCOptions.